File: baekjoon/bj1541.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class bj1541 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int result = 0;
        StringTokenizer st = new StringTokenizer(br.readLine(), "-");

        StringTokenizer plusSt = new StringTokenizer(st.nextToken(), "+");
        int plusToken = plusSt.countTokens();
        for (int i = 0; i < plusToken; i++) {
            result += Integer.parseInt(plusSt.nextToken());
        }
        
        while (st.hasMoreTokens()) {
            int temp = 0;
            StringTokenizer minusSt = new StringTokenizer(st.nextToken(), "+");
            while(minusSt.hasMoreElements()){
                temp += Integer.parseInt(minusSt.nextToken());
            }
            result -= temp;
        }
        System.out.println(result);
    }
}

File: baekjoon/bj1541.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine(), "-")
    var result: Int = 0;

    val plusst = StringTokenizer(st.nextToken(), "+")
    val stcount: Int = plusst.countTokens();
    for (i in 0 until stcount) {
        result += plusst.nextToken().toInt()
    }
    while (st.hasMoreTokens()) {
        var temp: Int = 0
        val minusSt=StringTokenizer(st.nextToken(),"+")
        while (minusSt.hasMoreTokens()){
            temp+=minusSt.nextToken().toInt()
        }
        result-=temp
    }
    println(result)

}